plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/DebugNativeAppEngineStandardProject.java

```diff
@@ -0,0 +1,152 @@
+
+    project = SwtBotAppEngineActions.createNativeWebAppProject(bot, "testapp", null,
+        "app.engine.test", "my-project-id");
+    assertTrue(project.exists());
+
+    SWTBotTreeItem project = SwtBotProjectActions.selectProject(bot, "testapp");
+    assertNotNull(project);
+    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, new Runnable() {
+      @Override
+      public void run() {
+        bot.menu("Run").menu("Debug As").menu("1 Debug on Server").click();
+      }
+    });
+
+    SwtBotTestingUtilities.clickButtonAndWaitForWindowChange(bot, bot.button("Finish"));
+
+    bot.perspectiveById("org.eclipse.debug.ui.DebugPerspective").activate(); // IDebugUIConstants.ID_DEBUG_PERSPECTIVE
+    SWTBotView debugView = bot.viewById("org.eclipse.debug.ui.DebugView"); // IDebugUIConstants.ID_DEBUG_VIEW
+    debugView.show();
+
+    SWTBotTree launchTree =
+        new SWTBotTree(bot.widget(widgetOfType(Tree.class), debugView.getWidget()));
+     SwtBotTreeUtilities.waitUntilTreeHasItems(bot, launchTree);
+    SWTBotTreeItem[] allItems = launchTree.getAllItems();
+    SwtBotTreeUtilities.waitUntilTreeHasText(bot, allItems[0]);
+    assertTrue("No App Engine launch found",
+        allItems[0].getText().contains("App Engine Standard at localhost"));
+
+    SWTBotView consoleView = bot.viewById("org.eclipse.ui.console.ConsoleView"); // IConsoleConstants.ID_CONSOLE_VIEW
+    consoleView.show();
+    assertTrue("App Engine console not active", consoleView.getViewReference()
+        .getContentDescription().contains("App Engine Standard at localhost"));
+    final SWTBotStyledText consoleContents =
+        new SWTBotStyledText(bot.widget(widgetOfType(StyledText.class), consoleView.getWidget()));
+    SwtBotTestingUtilities.waitUntilStyledTextContains(bot,
+        "Starting module \"default\" running at: http://localhost:8080", consoleContents);
+
+    assertEquals("Hello App Engine!",
+        getUrlContents(new URL("http://localhost:8080/hello"), SWTBotPreferences.TIMEOUT));
+
+    for (SWTBotToolbarButton b : consoleView.getToolbarButtons()) {
+      if ("Stop the server".equals(b.getToolTipText())) {
+        b.click();
+      }
+    }
+    SwtBotTreeUtilities.waitUntilTreeContainsText(bot, allItems[0], "<terminated>");
+    assertNoService(new URL("http://localhost:8080/hello"));
+  }
+
+
+  /**
+   * Check that there is no remote service for the URL.
+   */
+  private void assertNoService(URL url) {
+    try {
+      getUrlContents(url, 10);
+      fail("There appears to be life at " + url);
+    } catch (IOException ex) {
+      /* expected */
+    }
+  }
+
+
+  /**
+   * Read the content as a string from the specified URL.
+   * 
+   * @throws IOException if cannot connect or timeout
+   */
+  private String getUrlContents(URL url, long timeoutInMilliseconds) throws IOException {
+    StringBuilder content = new StringBuilder();
+    URLConnection conn = url.openConnection();
+    conn.setConnectTimeout(5); // ms
```

Comment:
should probably bump this to at least 100; 5 may not be long enough to detect a real running server